solidity/random-beacon/contracts/libraries/Groups.sol

```diff
@@ -14,17 +14,28 @@ library Groups {
 
     struct Group {
         bytes groupPubKey;
-        uint256 activationTimestamp;
+        uint256 activationBlockNumber;
```

Comment:
Why did we switch to blocks? I know it's how v1 operates but were there any other reasons for that? It feels that >= 2 weeks group lifetime is not prone to any miner manipulation and expressing group lifetime in seconds is more aligned with undelegation time and reward periods.

Reply:
The trigger was the calculation of a stale group status [here](https://github.com/keep-network/keep-core/blob/groups-expiration/solidity/random-beacon/contracts/libraries/Groups.sol#L216). Relay entry timeout is in blocks and I couldn't add sec + blocks. The result was changing activation timestamp and a group lifetime to blocks.
I think having everything in blocks might be less human-friendly but helps stay consistent across libraries and calculation purposes.